### PR TITLE
backupccl: delete reintroduced spans support

### PR DESCRIPTION
**Pull request by dt**
This was to handle tables that could not be backed up during imports but imports are mvcc now so this is no longer an issue.

Release note: none.
Epic: none.

### CODE REVIEW

File: pkg/ccl/backupccl/testdata/backup-restore/in-progress-import-rollback
Comment (msbutler) [inline pkg/ccl/backupccl/testdata/backup-restore/in-progress-import-rollback:1]: super nit: the comments in this test file should be updated. E.g. 
```
# Ensure incremental backups backup the newly online spans from ts=0, as the
# import was rolled back via non-mvcc clear range.
```

Also feel free to leave this for a follow up pr.

File: pkg/ccl/backupccl/targets.go [hunk 1]
```diff
@@ -443,84 +443,6 @@ func checkMissingIntroducedSpans(
 			}
 			tablesIntroduced[descpb.ID(tableID)] = struct{}{}
 		}
-
```
Comment (msbutler) [inline pkg/ccl/backupccl/targets.go:446]: i think we can get rid of this whole checkMissingIntroducedSpans function, right?

File: pkg/ccl/backupccl/restore_span_covering.go [hunk 2]
```diff
@@ -107,8 +107,7 @@ var _ backupManifestFileIterator = &sstFileIterator{}
 
 // createIntroducedSpanFrontier creates a span frontier that tracks the end time
 // of the latest incremental backup of each introduced span in the backup chain.
-// See ReintroducedSpans( ) for more information. Note: this function assumes
-// that manifests are sorted in increasing EndTime.
+// Note: this function assumes that manifests are sorted in increasing EndTime.
```
Comment (msbutler) [inline pkg/ccl/backupccl/restore_span_covering.go:110]: i believe we can remove this IntroducedSpanFrontier from restore as well, but perhaps we can deal with that in a follow up. I need to think a little more about the fact that 23.2 backups will still have reintroduced spans.

**Review by msbutler — APPROVED** [review]

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request merged**